thoth/adviser/run.py

```diff
@@ -73,13 +74,20 @@ def subprocess_run(
             result_dict.update(
                 dict(error=False, error_msg=None, report=report.to_dict())
             )
+        except AdviserRunException as exc:
+            _LOGGER.warning("Adviser run failed: %s", str(exc))
+            result_dict.update(dict(error=True, error_msg=str(exc), report=exc.to_dict()))
         except Exception as exc:
-            _LOGGER.exception("Adviser raised an exception: %s", str(exc))
-            result_dict.update(dict(error=True, error_msg=str(exc), report=None))
-            sys.exit(1)
+            _LOGGER.exception("Adviser raised exception: %s", str(exc))
+            result_dict.update(dict(
+                error=True,
+                error_msg=str(exc),
+                report=dict(ERROR="An error occurred, see logs for more info")
+            ))
 
+        # Always submit results, even on error.
         print_func(time.monotonic() - start_time, result_dict)
-        sys.exit(0)
+        sys.exit(int(result_dict["error"]))
```

Comment:
this will yield 1 on error?

Reply:
yes, `sys.exit(1)` on error so we will propagate failure (e.g. later schedule solvers based on an error in workflows)